sentry-core/src/main/java/io/sentry/core/SentryThreadFactory.java

```diff
@@ -79,7 +82,8 @@ public SentryThreadFactory(final @NotNull SentryStackTraceFactory sentryStackTra
       }
 
       for (Map.Entry<Thread, StackTraceElement[]> item : threads.entrySet()) {
-        result.add(getSentryThread(currentThread, item.getValue(), item.getKey()));
+        result.add(
+            getSentryThread(currentThread, item.getValue(), item.getKey(), mechanismThreadIds));
```

Comment:
I guess the only suggestion would be to remove this parameter and pass the decision, And lift the code that does the decision

```suggestion
            getSentryThread(isCrashed, item.getValue(), item.getKey()));
```